internal/service/alarms/internal/alertmanager/alertmanager.go

```diff
@@ -61,3 +66,103 @@ func Setup(ctx context.Context, cl client.Client) error {
 	slog.Info("Successfully configured alertmanager")
 	return nil
 }
+
+// ConvertAmToAlarmEventRecordModels get alarmEventRecords based on the alertmanager notification and AlarmDefinition
+func ConvertAmToAlarmEventRecordModels(am *api.AlertmanagerNotification, aDefinitionRecords []models.AlarmDefinition) []models.AlarmEventRecord {
```

Comment:
I would create a type labels map[string][string] and then have

func(r *labels) Get.....() {}

for each function you have. It's cleaner I think

Reply:
hmmm so so we are also dealing with 'annotation' (as well as 'label') later to combine them together for extension. Might be a bit strange if we do type labels but leave the rest as regular map. IMO this is already quite minimal? Otherwise maybe another PR for that change?

Reply:
that's another thing, why are we starting with all possible data in extensions? We should start empty and adding stuff later in my opinion

Reply:
Yeah its actually pretty valuable to have everything in there to actually understand the alerts that are in the DB (without it its simply UUIDs in the DB)